wagtail/admin/rich_text/editors/hallo.py

```diff
@@ -23,7 +23,12 @@ def construct_plugins_list(self, plugins):
 
     @property
     def media(self):
-        return Media(js=self.js, css=self.css)
+        js = [versioned_static(js_file) for js_file in self.js]
```

Comment:
A potential issue here:
Django's `Media` class has some conditional logic in place to [accept either a relative or absolute path](https://docs.djangoproject.com/en/2.2/topics/forms/media/#form-asset-paths) - relative paths will be passed through `static`, while absolute paths will be left unchanged. However, the `static` / `versioned_static` function itself does not include this check - passing an already-absolute path to `static` will mangle it.

In the current version of the code, we're only calling `versioned_static` on literal strings that we know are relative paths. However, by moving the `versioned_static` call here, we'll also be applying it to plugins defined by third-party code outside of Wagtail, and we can't be sure whether that third-party code is passing absolute or relative paths (since both are valid, as per the `Media` class's behaviour).

I think the best fix is to update `versioned_static` to replicate the `Media` behaviour, so that it leaves paths beginning with `/`, `http://` or `https://` unchanged. It feels slightly hacky, but there's already a bit of 'defensive' logic in place there (i.e. not adding the querystring if the result of `static` has one already) so it wouldn't be too much of a leap.